text/3346-t-opsem.md

```diff
@@ -0,0 +1,110 @@
+
+#### Examples
+
+ - **When may a raw pointer be used to write to memory that a unique reference also points to?**
+
+   Writing via a raw pointer requires `unsafe` code, meaning this question is in scope for T-opsem. The answer to this question also has broad implications for the usability of the `unsafe` subset of the Rust language. As such, the lang team will need to approve the high-level answer.
+   
+ - **Do match guards have semantic meaning?**
+
+   Match guards are inserted by the compiler around match statements to ensure that if guards cannot change the value being matched on. Whether match guards should exist at all primarily affects exhaustiveness checking, and so is a question for T-types and T-lang to answer, not T-opsem.
+   
+   However, T-opsem is responsible for deciding whether these match guards have semantic meaning at runtime, as that is only observable to `unsafe` code in if-guards.
+   
+ - **Should an `Unordered` atomic ordering be added to the language?**
+
+   The behavior of an `Unordered` ordering is distinguishable from a `Relaxed` ordering in strictly safe code. However, T-lang should still consult T-opsem on this question, because T-opsem is expected to be the team that is most familiar with and has the most interest in the semantics of atomic memory models.
+
+## Relationships to other teams
+
+**T-lang**: The team is a subteam of T-lang. It has the same relationship to T-lang as T-types has. This means decisions about "details" will be made by the team alone, but decisions around the big picture "direction" will require consultation with T-lang.
+
+**T-types**: As T-types will no longer own semantics questions, the responsibilities of T-opsem and T-types are not expected to overlap. However, like other teams, T-types is expected to consult T-opsem on any changes that require support from the operational semantics. For example, if T-types wants to extend the borrow checker to allow more code patterns, T-opsem must confirm that the code that this permits can be supported by a reasonable operational semantics.
+
+**T-compiler**: Unlike T-types, T-opsem is not a subteam of T-compiler as it does not own any implementations. However, T-compiler is still expected to request approval from T-opsem before adding any optimization that depends on new theorems about the operational semantics. T-opsem will ensure that these theorems are expected to be true and are reasonable things for the compiler to depend on now.
```

Comment:
The two teams are expected to heavily interact on the syntax and semantics of MIR specifically -- right? that seems worth documenting.

Reply:
I've included this for T-types. I'm somewhat hesitant to include this for T-compiler, since I don't see a fundamental reason that rustc's MIR should be the same as the spec's MIR

Reply:
> I don't see a fundamental reason that rustc's MIR should be the same as the spec's MIR

If they are not the same, we will probably want to have a translation from rustc's MIR to the spec's core language, as a means to give precise semantics to rustc's MIR.